### ISSUE TITLE
Optimize Windows .exe icon

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#803]
dist/artwork/dusty-app.ico is 260 KiB. That's pretty big. Let's shrink it to bring the .exe file size down.

**Comment by rol1510** [comment]
By opening the dusty-app.ico file in GIMP and exporting it again, using the png compression for the 256x256 image, I got the following results:

(RELEASE Build; VS 2022 on Windows 10)
```
BEFORE
exe:	969 KB (992 768 Bytes)
icon:	269 KB (275 822 Bytes)

AFTER
exe:	728 KB (745 984 Bytes)	saved 24.9%
icon:	28,7 KB (29 406 Bytes)  saved 89.3%
```

I could not find any information on how widely supported the compression inside .ico files is, but GIMP issues a warning, so maybe don't compress the smaller images?
By also compressing the 32x32 and 16x16 images we can gain another 3kB.

From what I saw, `icotool` doesn't support compressing the images, but are ways to do it without depending on gimp.


Gimp Settings used:
![image](https://user-images.githubusercontent.com/29011024/181002762-dd466c87-9208-4e48-afe4-5c15c22c22a8.png)
